witness.schema.json

```diff
@@ -0,0 +1,393 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$defs": {
+    "format-version": {
+      "description": "Version of the format.",
+      "const": "0.1"
```

Comment:
Any particular copy of the schema specifies a single version of the format though, not multiple incompatible versions. And the only reason to bump the version would be due to some (notable) incompatible changes to the schema.

Although this version is specified per-entry, so it's also unclear how the versioning is supposed to work. Is a single witness file allowed to contain entries with different versions, each one cohering to the respective version of the schema? If so, then that significantly complicates everything (beyond just this schema definition).
Since it's `format-version`, not `entry-type-version`, then the more reasonable thing is that all entries use the same version and for each version there could be a schema.

Reply:
That's a good point. I agree that allowing entries that adhere to different format versions would complicate things too much, but of course not doing so still has several advantages, most notably:
1. Validators will either have to be able to produce `loop_invariant_certificate` and similar entries in each format version that they support as input (potentially leading to unnecessary code duplication), or will have to able to transcribe (parts of) an old witness into a new format version. Otherwise we end up with validators for specific versions of the format.
2. Similarly, old witnesses would have to be transcribed to be of use to tools that only support newer versions of the format
3. Each new format version would require a separate copy of the schema that needs to be maintained (although only the latest one, if any, should subject to change)

I guess this is something that has to be discussed further, but should not be blocking for this PR, especially since the old README (though not the old schema) already had this problem.